### ISSUE TITLE
Notification for files I changed myself

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/client#2072]
Oftentimes I change files from my mobile phone or from the web interface. Then I get notifications in the desktop client about these changes.

![screenshot from 2014-08-15 19 01 26](https://cloud.githubusercontent.com/assets/925062/3949660/184e759e-26ba-11e4-9e52-5e790c252ecc.png)

Instead, the desktop client should only show notifications about changes which were made by others. So basically only monitor shared folders/files.

@danimo @dragotin?

**Comment by Dianafg76** [comment]
@jancborchardt, @guruz Can this be closed or is there still work to be done?

**Comment by dragotin** [comment]
The decision is pending. The old struggle. Let me throw in that we could have a config option saying 

* [ ] notify about changes I did myself on other machines

**Comment by jancborchardt** [comment]
> we could have a config option saying

You’re seriously kidding me, right? :D

Homies: Notifications should be reserved for the really important things. Stuff I did _myself_ is not important and not worthy to be notified of, as I know it already! We need to cut down notifications wherever possible, so the important notifications don’t drown in a sea of irrelevance.

**Comment by dragotin** [comment]
Hmm, I am still against your proposal of not showing the myself-notifications.

**Comment by guruz** [comment]
Getting a notification to say "nice, it really synced from my Phone to my Computer" __is__ an important notification.

**Comment by jancborchardt** [comment]
@dragotin why? Please provide arguments.

@guruz lol – are you serious? People _expect_ this. If ownCloud would be version 0.1, I would maybe understand. But this is supposed to be stable software. Notifications should only be shown for stuff which is out of the ordinary (files shared with you, errors), but _absolutely not_ for the most basic thing a sync client should do.

**Comment by moscicki** [comment]
Why not the option? Some people want to get notified about their own activity and some other do not. Where is a problem?

**Comment by jancborchardt** [comment]
Because if we introduce options for every little thing such as this, the truly important options will not stand out anymore. In open source it’s a typical thing to just ask for an option – but with ownCloud we make proper design decisions instead of offloading them to every single user.

**Comment by moscicki** [comment]
That's correct. On the other hand maybe not every single user has the same needs. So maybe the single "proper design decision" does not simply exist. It could be possible to split basic options and advanced options, couldn't it?

**Comment by jancborchardt** [comment]
Yeah, but not in this case. We are only talking about this because it’s how it’s been so far – no other sync system does it like this and it simply is unexpected behavior. There’s nothing to introduce a setting for, this is just something we need to fix.

**Comment by moscicki** [comment]
Okey. "no other sync system does it like this " -- isn't owncloud supposed to be better? ;-) Anyway, I don't think this is critical.

**Comment by jancborchardt** [comment]
> isn't owncloud supposed to be better?

Exactly. And having notifications for your own actions makes ownCloud _worse_ because it’s annoying. :)

**Comment by tflidd** [comment]
The other way round would be nice: e.g. if the sync of your pictures from your phone fails, you get a notification on your desktop ;-) But that's hard to catch what really failed, what failed temporarily, ....
When you use certs to authenticate oc clients, you could provide a list of synced device on the OC interface, with the last connection and the last status of each client. But that's far beyond this topic :D

Back to the client's notification: If you don't trust 100% the sync processes, this quick positive feedback is really nice. You don't have to click and the message disappears automatically. I'm not changing device so often that it would be annoying. Perhaps schedule the removal to a later release when the sync process of all clients is more reliable (especially on the mobile ones).

**Comment by jancborchardt** [comment]
@guruz @dragotin can we schedule this to a milestone? It’s really a pretty annoying misuse of notifications for something which is not important. We can have a call about it if needed.

**Comment by dragotin** [comment]
@jancborchardt we still do not seem to have a common understanding of what we want :-)

Also, I think the the whole topic *notification* is going further than what we currently have. Since there are limitations of the notification systems across all platforms we do not have an easy way to do interactive notifications. That is requested in owncloud/activity#349 tough. So we have to come up with something "bigger" anyway. 

Some thoughts about that:
* We need interactive notifications with links and Yes/No.
* We need a way to "filter" or organize on certain levels of information, ie. "self caused" sync info 
* We need tie together the server activities and the client sync activities
* We need to handle certain client error displays such as soft-errors, ignores and stuff
* We want to display overall sync states of sync connections.

All that reminds me on #1748 Maybe we want to start with using QML on that.

@jancborchardt I'd be happy if you could be creative on that. I think the solution we're striving for should not only consider the simply usecase with hiding all info, but come up with an interface that is clever enough to hide as much as needed but offers all that is wanted by more than one user group.

**Comment by phil-davis** [comment]
I know that techo users request lots of configuration options on software, but I think this is a case where an option is worthwhile. Other utilities that interact between multiple systems do this sort of thing - e.g. in the cloud file sharing space, Dropbox has "Show desktop notifications", in Skype you can let it tell you about contacts that come online or you can shut it up. In TeamViewer you can let it tell you about computers that come on and off line or you can shut it up.
In the same vein, ownCloud could let the user choose to see "information" notifications about "routine" stuff, or not.
You can go the whole hog and have a multi-level info/warning/error/fatal series of message levels and let the user choose how much they want to see, or you can just have 2 levels "info" and "everything else".
And as @dragotin says, you may or may not want categories of "info" level - self-caused info, other-caused info...

**Comment by jancborchardt** [comment]
> in the cloud file sharing space, Dropbox has "Show desktop notifications"

We offer the same setting in the General tab of the settings.

@dragotin let’s not get off track here. Sure, all this stuff would be cool and exciting but the only problem I have with notifications right now is there is a group of irrelevant ones being triggered – stuff you change yourself, on any device.

**Comment by dragotin** [comment]
@jancborchardt well, problem is that ....that's your opinion. Others have expressed other opinions here in this thread, and I do not think its obvious.

**Comment by jancborchardt** [comment]
So apparently the problem here is that we don’t know on the client if the file was changed by yourself or by another user. That results in every change being a notification.

@DeepDiver1975 do we have that info somewhere or can we get it to the client without a crazy performance hit? Does the activity stream get us that info?

@dragotin @ogoffart I really think this is an important topic. Cluttering the notifications with irrelevant stuff you already know makes them basically useless. It was never part of the intended design to show notifications for every change you made yourself. (It might be interesting for you devs to know when something synced, but normal people just expect that.) cc @MTRichards
